server/server.go

```diff
@@ -943,3 +948,68 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startReloadWatcherForMirrorRegistries starts a new goroutine that
+// monitors the "registries.conf.d" directory. If not set, it defaults to
+// "/etc/containers/registries.conf.d". The routine reloads the registries
+// configuration when a file is modified, created or deleted in the directory.
+func (s *Server) startReloadWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, defaulting to /etc/containers/registries.conf.d")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf(ctx, "Failed to create new watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	log.Infof(ctx, "Registered reload watcher for mirror registries configuration")
+
+	s.reloadMirrorRegistriesConfiguration(ctx, watcher, registriesConfDDir)
+}
+
+func (s *Server) reloadMirrorRegistriesConfiguration(ctx context.Context, watcher *fsnotify.Watcher, path string) {
+	if err := watcher.Add(path); err != nil {
+		log.Errorf(ctx, "Failed to add watcher for path %q: %s", path, err)
+		return
+	}
+
+	reloadLock := make(chan struct{}, 1)
+	reloadLock <- struct{}{}
```

Comment:
This might not be _quite_ the approach? If there is a bust of activity, generating 5 events, AFAICS this going to result in 5 reloads, over 5*`eventStaggerDuration`.

I’d expect the loop to look for changes to “settle”, e.g. by arming/re-arming a timer on every change, and triggering a reload if the timer fires without any more changes arriving in the meantime.